lib/liquid/context.rb

```diff
@@ -87,9 +83,9 @@ def invoke(method, *args)
     end
 
     # Push new local scope on the stack. use <tt>Context#stack</tt> instead
-    def push(new_scope = {})
-      @scopes.unshift(new_scope)
-      raise StackLevelError, "Nesting too deep".freeze if @scopes.length > Block::MAX_DEPTH
+    def push
+      @scopes.unshift({})
+      raise StackLevelError, "Nesting too deep".freeze if @scopes.length > (Block::MAX_DEPTH + 1)
```

Comment:
why the +1?

Reply:
This is allowing for root scope now that it will be included even if its empty